tests/python_functional/src/common/file.py

```diff
@@ -22,33 +22,84 @@
 #############################################################################
 import logging
 
+from pathlib2 import Path
+
+from src.common.blocking import DEFAULT_TIMEOUT
 from src.common.blocking import wait_until_true
+from src.common.blocking import wait_until_true_custom
 from src.common.operations import open_file
 
 logger = logging.getLogger(__name__)
 
 
 class File(object):
     def __init__(self, file_path):
-        self.__file_path = file_path
+        self.path = Path(file_path)
         self.__opened_file = None
 
     def __del__(self):
-        if self.__opened_file:
-            self.__opened_file.close()
-            self.__opened_file = None
-
-    def __is_file_exist(self):
-        return self.__file_path.exists()
+        if self.is_opened():
+            self.close()
 
     def wait_for_creation(self):
-        file_created = wait_until_true(self.__is_file_exist)
+        file_created = wait_until_true(self.path.exists)
         if file_created:
-            logger.debug("File has been created:\n{}".format(self.__file_path))
+            logger.debug("File has been created: {}".format(self.path))
         else:
-            logger.debug("File not created:\n{}".format(self.__file_path))
+            raise Exception("File was not created in time: {}".format(self.path))
         return file_created
 
     def open(self, mode):
-        self.__opened_file = open_file(self.__file_path, mode)
+        self.__opened_file = open_file(self.path, mode)
         return self.__opened_file
+
+    def close(self):
+        self.__opened_file.close()
+        self.__opened_file = None
+
+    def is_opened(self):
+        return self.__opened_file is not None
+
+    def readline(self):
+        if not self.is_opened():
+            raise Exception("File was not opened before trying to read from it.")
+        return self.__opened_file.readline()
+
+    def read(self):
+        content = ""
+        while True:
+            buffer = self.readline()
```

Comment:
Thanks for the review comment. 
I have checked this and it works with read() also. So I will fix this to use read().
As I remember there was an other place where we should use readline() instead of read() similar to your previous finding in
https://github.com/syslog-ng/syslog-ng/commit/431bc3583c25842c6c4155230dfeb52787191926